compiler/rustc_middle/src/ty/util.rs

```diff
@@ -252,7 +265,7 @@ impl<'tcx> TyCtxt<'tcx> {
 
                 ty::Alias(..) => {
                     let normalized = normalize(ty);
-                    if ty == normalized {
+                    if ty == normalized || normalized.references_error() {
```

Comment:
Why is this necessary?

Reply:
It makes us return immediately when there's an error instead of recursing further. Although it saves just one iteration so I could perhaps remove it.

Reply:
Please remove it if it's unnecessary; adding extra logic just makes the function harder to understand IMO

Reply:
I agree. Removed now.